projects/step-frontend/src/lib/modules/timeseries/util/timeseries-colors-pool.ts

```diff
@@ -51,11 +52,16 @@ export class TimeseriesColorsPool {
    */
   getStatusColor(status: string): string {
     if (!status) {
-      let randomColor = this.randomRGBA();
-      statusColors[status.toLowerCase()] = randomColor;
-      return randomColor;
+      let grey = '#cccccc';
+      statusColors[this.NO_STATUS] = grey;
+      return grey;
```

Comment:
You are setting something from within the setter. I think a cleaner solution would be to set the colour statusColors[this.NO_STATUS] = grey; in an init method and just return statusColors[this.NO_STATUS].

Also note that grey should be a const here since it is not changing.

Reply:
agree